dbt/include/redshift/macros/adapters/apply_grants.sql

```diff
@@ -24,4 +62,72 @@ where has_table_privilege(u.usename, '{{ relation }}', privilege_type)
     and u.usename != current_user
     and not u.usesuper
 
+union all
+-- check that group has table privilege
+select
+    'group' as grantee_type,
+    g.groname as grantee,
+    p.privilege_type
+from pg_group g
+cross join privileges p
+where exists(
+    select *
+    from information_schema.table_privileges tp
+    where tp.grantee=g.groname
+    and tp.table_schema=replace(split_part('{{ relation }}', '.', 2), '"', '')
+    and tp.table_name=replace(split_part('{{ relation }}', '.', 3), '"', '')
+    and LOWER(tp.privilege_type)=p.privilege_type
+)
+
+union all
+-- check that role has table privilege
+select
+    'role' as grantee_type,
+    r.role_name as grantee,
+    p.privilege_type
+from svv_roles r
+cross join privileges p
+where exists(
+    select *
+    from svv_relation_privileges rp
+    where rp.identity_name=r.role_name
+    and rp.namespace_name=replace(split_part('{{ relation }}', '.', 2), '"', '')
+    and rp.relation_name=replace(split_part('{{ relation }}', '.', 3), '"', '')
```

Comment:
We should use [`Relation` class attributes](https://docs.getdbt.com/reference/dbt-classes#using-relations) (like [here](https://github.com/dbt-labs/dbt-postgres/blob/7a4c0219fee0da0b6dcc086929e3985601f8d82a/dbt/include/postgres/macros/adapters.sql#L222-L223) and [here](https://github.com/dbt-labs/dbt-bigquery/blob/2c77893d0fb7b5d76ba85cc16712b9563a60329b/dbt/include/bigquery/macros/adapters/apply_grants.sql#L7-L8)) rather than splitting on dot separators.

```suggestion
    and rp.namespace_name = '{{ relation.schema }}'
    and rp.relation_name = '{{ relation.identifier }}'
```